### PR TITLE
Fixes #188

### LINKED ISSUE #188: Bad E/Z assignment from ctab
```python
In [15]: print mb
CHEBI:20745
  Marvin  03051216372D          

 22 22  0  0  0  0            999 V2000
   -3.1832    2.5375    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -3.1832    1.7125    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.7543    1.7125    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.7543    2.5375    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -2.4687    2.9500    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -2.4687    3.7750    0.0000 Cl  0  0  0  0  0  0  0  0  0  0  0  0
   -2.4687    1.3000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -2.4687    0.4750    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -3.8977    0.4750    0.0000 Cl  0  0  0  0  0  0  0  0  0  0  0  0
   -3.1832    0.0625    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -3.1832   -0.7625    0.0000 Cl  0  0  0  0  0  0  0  0  0  0  0  0
   -1.0398    0.4750    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
   -1.7543    0.0625    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.7543   -0.7625    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.0398   -1.1750    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.3253    0.0625    0.0000 Cl  0  0  0  0  0  0  0  0  0  0  0  0
   -0.3253   -0.7625    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.3891   -1.1750    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.1036   -0.7625    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
    1.1036   -2.4125    0.0000 O   0  5  0  0  0  0  0  0  0  0  0  0
    0.3891   -2.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.3253   -2.4125    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
  5  1  1  0  0  0  0
  5  4  2  0  0  0  0
  1  2  2  0  0  0  0
  2  7  1  0  0  0  0
  7  3  2  0  0  0  0
  3  4  1  0  0  0  0
  5  6  1  0  0  0  0
  7  8  1  0  0  0  0
  8 10  2  0  0  0  0
  8 13  1  0  0  0  0
 10  9  1  0  0  0  0
 10 11  1  0  0  0  0
 13 12  2  0  0  0  0
 13 14  1  0  0  0  0
 14 15  2  0  0  0  0
 15 17  1  0  0  0  0
 17 16  1  0  0  0  0
 17 18  2  0  0  0  0
 18 21  1  0  0  0  0
 18 19  1  0  0  0  0
 21 20  1  0  0  0  0
 21 22  2  0  0  0  0
M  CHG  1  20  -1
M  END
In [9]: m = Chem.MolFromMolBlock(mb)

In [10]: for b in m.GetBonds():
    if b.GetBondType()==Chem.BondType.DOUBLE:
        print b.GetStereo()
   ....:         
STEREONONE
STEREONONE
STEREOE
STEREOZ
STEREONONE

In [11]: m2 = Chem.MolFromMolBlock(Chem.MolToMolBlock(m))

In [12]: for b in m2.GetBonds():
    if b.GetBondType()==Chem.BondType.DOUBLE:
        print b.GetStereo()
   ....:         
STEREOANY
STEREONONE
STEREOZ
STEREOZ
STEREONONE

In [13]: m2 = Chem.MolFromMolBlock(Chem.MolToMolBlock(m2))

In [14]: for b in m2.GetBonds():
    if b.GetBondType()==Chem.BondType.DOUBLE:
        print b.GetStereo()
   ....:         
STEREOANY
STEREONONE
STEREOZ
STEREOZ
STEREONONE
```

### PR DESCRIPTION
**Pull request by greglandrum**
This is a pretty straightforward one.
The bug was in the setting of single bond directions around double bonds.
The code that handles chained double bonds (i.e. things like `C/C=C/C=C/C=C/C`) and makes sure that the directions are consistent was not properly moving on to set dependent bonds after updating one side of a double bond.

### CODE REVIEW

**Comment by greglandrum** [comment]
Missed a case. Closing for the moment.

**Comment by bp-kelley** [comment]
The changes seem straight forward so I'm going to merge this tomorrow.  I will spend some time trying to understand this a bit better, I want to get a better feel for the performance impact of this code ( I suspect it is small, but current cip and stereoE/Z perception are a large chunk of the perception run time)

File: Code/GraphMol/FileParsers/MolFileStereochem.cpp [hunk 1]
```diff
@@ -729,20 +730,25 @@ bool isLinearArrangement(const RDGeom::Point3D &v1, const RDGeom::Point3D &v2,
 
 void updateDoubleBondNeighbors(ROMol &mol, Bond *dblBond, const Conformer *conf,
                                boost::dynamic_bitset<> &needsDir,
-                               std::vector<unsigned int> &singleBondCounts) {
+                               std::vector<unsigned int> &singleBondCounts,
```
Comment (DoliathGavid) [inline Code/GraphMol/FileParsers/MolFileStereochem.cpp]: can singleBondCounts be const here?

**Comment by DoliathGavid** [comment]
I looked at this as well. I was at first initially worried about the extra recursion in updateDoubleBondNeighbors, but even so it should still be at most O(# of bonds).

**Comment by greglandrum** [comment]
Given that it previously had to hit all of the double bonds too, I don't think the performance of this should be substantially worse than what we were seeing before.

@bp-kelley : you are absolutely right that the CIP and E/Z assignment performance is crap. @NadineSchneider and I have talked about this before and have some ideas about improving it, but that's definitely real work and would need to wait until the next release.

**Comment by bp-kelley** [comment]
@greglandrum It's not crap per-se, but it has an enormous run-time.  We've talked before about changing bond e/z and cip stereo to "on-demand" so some standard use cases (substructure searching etc) wouldn't need it to be computed by default.  This would make the smiles->parse->sanitize twice as fast.

The difficulty is that a lot of the code uses cip and e/z identification, so this might not really be possible.

**Comment by greglandrum** [comment]
The lazy approach is certainly possible, and I think it's the right thing
to do (along with improving the existing algorithm). It's just going to
take some thought and careful testing

**Comment by greglandrum** [comment]
I'd like to go ahead and create the release branch (because April 1 is just *the* day to do such things), so I'm going to merge this on my own based on the comments above.

Worst case we can fix things with a patch.

**Pull request merged**